packages/nimble-components/src/table/specs/cell-state-when-scrolling-hld.md

```diff
@@ -0,0 +1,111 @@
+# Nimble Table: Handling Cell State when Scrolling
+
+## Problem Statement
+
+Since the Nimble Table uses virtualization, if the table contains more rows than can be shown in the current table viewport, we will re-use the DOM elements / controls within the table cells as the user scrolls through the table. This is handled by the TanStack virtualizer determining what row indexes are visible at the current scroll position, and then we re-bind the table cell views to the new cell state (including the row data). We need to ensure that cell state isn't incorrectly preserved/ applied to the cells once they represent different row values.
+
+Some of the cell state that would incorrectly apply to new rows (once the user scrolls) is:
+
+-   Selected text (which can span multiple rows), for both editable and non-editable cells
+-   Text cursor position (for editable controls in table cells)
+-   Pending input (for editable controls in table cells)
+-   Open popups (action menu or select/combobox cells whose popup is open)
+
+## Links To Relevant Work Items and Reference Material
+
+[#997: Table virtualization: Clear/Commit (and potentially save/restore) row/cell state when scrolling](https://github.com/ni/nimble/issues/997)
+
+## Implementation / Design
+
+When the user scrolls the table, we will:
+
+-   'Blur' (Lose focus) from any active/focused control in a table cell. Generally this means that whatever text the user began to change in that control will be committed, not discarded.
+-   If a cell action menu is open (in which case a menu item is focused), we'll close the associated menu via the menu button.
+
+We will not:
+
+-   Detect and deselect selected table text on scroll. This means that if table text is selected and the user scrolls, text will still be selected (potentially different text). We don't consider that to be a primary table interaction, and it shouldn't harm functionality (it's easy to resolve for the user by clicking off to deselect text). It's also the same behavior as the current `sl-grid`/`smart-table` used in SystemLink Enterprise.
```

Comment:
In the prototype, selected text within a text field is deselected. I'm guessing that's a side effect of calling blur, not something that will happen for non-input text? If so it might be good to clarify that in this description.

Reply:
Correct, that's a result of calling blur() so it'll only be for text in editable controls. I'll update the HLD.

Reply:
I revamped the prototype to use focusable-but-not-editable text. Hopefully that'll simplify things a bit, and remove the issues/questions related to editable controls and updating the table value that aren't directly related to this HLD.